onnxruntime/python/onnxruntime_pybind_state_common.h

```diff
@@ -410,7 +410,7 @@ bool CheckIfTensor(const std::vector<const NodeArg*>& def_list,
                    const std::string& name,
                    /*out*/ ONNX_NAMESPACE::TypeProto& type_proto);
 
-#ifdef ENABLE_TRAINING
+#ifdef ENABLE_DLPACK
```

Comment:
If ENABLE_DLPACK is on when ENABLE_TRAINING is, that's ok but I did not see anything related to that so I would write something ``#if defined(ENABLE_DLPACK) || defined(ENABLE_TRAINING)`` everywhere it is needed. You also need to expose the option in build.py (``--enable_dlpack``).

Reply:
That seems more safer. Will fix